iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -38,7 +38,7 @@ public AmqpConnectionHolder(IConnectionCredentials connectionCredentials, IotHub
             AdditionalClientInformation additionalClientInformation,
             IotHubClientAmqpSettings amqpSettings,
             Func<DirectMethodRequest, Task> onMethodCallback,
-            Action<AmqpMessage, string, IotHubClientException> twinMessageListener,
+            Func<AmqpMessage, string, IotHubClientException, Task> twinMessageListener,
```

Comment:
We need the twinMessageListener to be async so we can call the DeserializeAsync method that takes a stream